satip/download.py

```diff
@@ -0,0 +1,303 @@
+        else eumetsat_cloud_name_to_datetime
+    )
+    # Check all the right size
+    satpy_reader = "seviri_l1b_native" if product_id == RSS_ID else "seviri_l2_grib"
+
+    if product_id == RSS_ID:
+        for f in new_files:
+            try:
+                file_size = eumetsat.get_filesize_megabytes(f)
+                if not math.isclose(file_size, NATIVE_FILESIZE_MB, abs_tol=1):
+                    print("Wrong Size")
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore", UserWarning)
+                    scene = Scene(filenames=[f], reader=satpy_reader)
+                    scene.load(SAT_VARIABLE_NAMES)
+                # Now that the file has been checked and can be open, move it to the final directory
+                base_name = get_basename(f)
+                file_date = date_func(base_name)
+                # Want to move it 1 minute in the future to correct the difference
+                file_date = file_date + timedelta(minutes=1)
+                if not fs.exists(os.path.join(directory, file_date.strftime(format="%Y/%m/%d"))):
+                    fs.mkdir(os.path.join(directory, file_date.strftime(format="%Y/%m/%d")))
+                fs.move(
+                    f, os.path.join(directory, file_date.strftime(format="%Y/%m/%d"), base_name)
+                )
+            except:
+                # Something is wrong with the file, redownload later
+                fs.rm(f)
+    else:
+        for f in new_files:
+            base_name = get_basename(f)
+            file_date = date_func(base_name)
+            file_size = eumetsat.get_filesize_megabytes(f)
+            if not math.isclose(file_size, CLOUD_FILESIZE_MB, abs_tol=1):
+                # Removes if not the right size
+                fs.rm(f)
+            else:
+                if not fs.exists(os.path.join(directory, file_date.strftime(format="%Y/%m/%d"))):
+                    fs.mkdir(os.path.join(directory, file_date.strftime(format="%Y/%m/%d")))
+                # Only move if the correct size
+                fs.move(
+                    f, os.path.join(directory, file_date.strftime(format="%Y/%m/%d"), base_name)
+                )
+
+    return
+
+
+def determine_datetimes_to_download_files(
+    directory,
+    start_date,
+    end_date,
+    product_id,
+) -> List[Tuple[datetime, datetime]]:
+    """
+    Check the given directory, and sub-directories, for all downloaded files.
+
+    Args:
+        directory: The top-level directory to check in
+        start_date:
+        end_date:
+        product_id:
+
+    Returns:
+        List of tuples of datetimes giving the ranges of time to download
+
+    """
+
+    pattern = "*.nat" if product_id == RSS_ID else "*.grb"
+    # Get all days from start_date to end_date
+    day_split = pd.date_range(start_date, end_date, freq="D")
+
+    # Go through files and get all examples in each
+    fs = fsspec.filesystem("file")  # TODO Update for other ones?
```

Comment:
Same idea as above could be applied here

```suggestion
    fs = fsspec.open(directory).fs
```

Reply:
Done!